docs/serializer.md

```diff
@@ -0,0 +1,171 @@
+* `create_bitangents_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_colors_cached(gvk::serializer& aSerializer, ...)`
+* `create_colors_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_weights_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_weights_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_weights_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_for_single_target_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_for_single_target_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_flipped_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_flipped_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_3d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_3d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `convert_for_gpu_usage_cached(gvk::serializer& aSerializer, ...)`
+
+# How to use
+The serializer is available via `gvk::serializer` class and can be used for, i.e. improving load
+times of huge orca scenes. It is initialized either by a `gvk::serializer::serialize` or a
+`gvk::serializer::deserialize` object, where both take a path to a **cache** file.
+`gvk::serializer::serialize` creates and writes to the **cache** file and
+`gvk::serializer::deserialize` reads from the **cache** file at the given path. A helper function
+`gvk::does_cache_file_exist(const std::string_view)` can be used to determine which initialization
+method shall be used.
+```
+const std::string cacheFilePath(pathToOrcaScene + ".cache");
+
+auto serializer = gvk::does_cache_file_exist(cacheFilePath) ?
+	gvk::serializer(gvk::serializer::deserialize(cacheFilePath)) :
+	gvk::serializer(gvk::serializer::serialize(cacheFilePath));
+```
+After initialization the current **mode** of the serializer can be retrived by calling `mode()` on
+the serializer, which either returns **serializer::mode::serialize** or
+**serializer::mode::deserialize**. To serialize or deserialize a type, the `archive(Type&&)`
+function template is called on the serializer object. This function serializes the given object to
+the **cache** file if the mode of the serialize is **serialize** or deserializes the object from the
+**cache** file into the given object if the mode is **deserialize**. The following code shows how a
+typical serializer usage can look like when used during orca scene loading.
```

Comment:
There should be a more detailed explaination about what exactly we are serializing: I.e. a `std::vector<vec3>` and then a `size_t`. 

And a question arises: Why serialize the vector first, and then the size? 
Or actually: Why do we need to serialize both? Is `positions.size()` not enough? Do we really need to serialize `numPositions` in addition?